src/commands/update.rs

```diff
@@ -1,11 +1,321 @@
+) -> Vec<DependencyChange> {
+    let added_packages = new_dependencies.difference(old_dependencies);
+    let removed_packages = old_dependencies.difference(new_dependencies);
+    let changed_dependencies: BTreeSet<&PackageName> = added_packages
+        .clone()
+        .chain(removed_packages.clone())
+        .map(|package| package.name())
+        .collect();
+
+    let mut dependency = Vec::new();
+
+    for dependency_name in changed_dependencies {
+        let matching_packages_removed = removed_packages
+            .clone()
+            .filter(|x| *x.name() == *dependency_name);
+        let matching_packages_added = added_packages
+            .clone()
+            .filter(|x| *x.name() == *dependency_name);
+
+        match (
+            matching_packages_added.clone().count(),
+            matching_packages_removed.clone().count(),
+        ) {
+            (1, 1) => {
+                // We know for certain that there is only one item in the iterator.
+                let package_added = matching_packages_added.last().unwrap();
+                let package_removed = matching_packages_removed.last().unwrap();
+
+                if package_added.gt(package_removed) {
+                    dependency.push(DependencyChange::Updated {
+                        from: package_removed.clone(),
+                        to: package_added.clone(),
+                    });
+                } else {
+                    dependency.push(DependencyChange::Downgrade {
+                        from: package_added.clone(),
+                        to: package_removed.clone(),
+                    });
+                }
+            }
+            (0, 1) => {
+                // We know for certain that there is only one item in the iterator.
+                let package_removed = matching_packages_removed.last().unwrap();
+                dependency.push(DependencyChange::Removed(package_removed.clone()));
+            }
+            (1, 0) => {
+                // We know for certain that there is only one item in the iterator.
+                let package_added = matching_packages_added.last().unwrap();
+                dependency.push(DependencyChange::Added(package_added.clone()));
+            }
+            (0, 0) => panic!("Impossible for the package name {} to not be removed or added if found in earlier.", dependency_name),
+            (_, _) => {
+                dependency.extend(matching_packages_added.map(|package| DependencyChange::Added(package.clone())));
+                dependency.extend(matching_packages_removed.map(|package| DependencyChange::Removed(package.clone())));
+            }
+        }
+    }
+
+    dependency
+}
+
+fn render_update_difference(dependency_changes: &[DependencyChange]) {
+    if dependency_changes.is_empty() {
+        return println!(
+            "{}No Dependency changes{}",
+            SetForegroundColor(Color::DarkGreen),
+            SetForegroundColor(Color::Reset)
+        );
```

Comment:
Nitpick: This not aligning annoys me! I prefer this. Up to you to add this or keep it how it is nbd.

```suggestion
        return println!(
            "{}{} No changed dependencies{}{}",
            SetAttribute(Attribute::Bold),
            SetForegroundColor(Color::DarkGreen),
            SetForegroundColor(Color::Reset),
            SetAttribute(Attribute::Reset)
        );
```

Reply:
Fixed lol